lib/osf-components/addon/components/storage-provider-manager/osf-storage-manager/component.ts

```diff
@@ -118,4 +123,31 @@ export default class OsfStorageManager extends Component<Args> {
         this.currentPage += 1;
         taskFor(this.getCurrentFolderItems).perform();
     }
+
+    @action
+    selectFile(file: OsfStorageFile, event: PointerEvent) {
```

Comment:
This function call is very nicely done. I would also add a try/catch block as well as error handling for no files present  or a non-file object type being returned.